File: 2022/01/part-one.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func readFile() int {
	inputFile, err := os.Open("./input")
	if err != nil {
		panic(err)
	}
	scanner := bufio.NewScanner(inputFile)
	max := 0
	sum := 0
	for scanner.Scan() {
		line := scanner.Text()
		if line == "" {
			if sum > max {
				max = sum
			}
			sum = 0
		} else {
			number, err := strconv.Atoi(line)
			if err != nil {
				panic(err)
			}
			sum += number
		}

	}
	return max
}

func main() {
	max := readFile()
	fmt.Println(max)
}

File: 2022/01/part-one.js
const fs = require("fs");

const data = fs.readFileSync("./input", "utf8");
const array = data.split("\n");

let max = 0;
let sum = 0;
for (let index = 0; index < array.length; index++) {
  const line = array[index];
  if (line == "") {
    if (sum > max) {
      max = sum;
    }
    sum = 0;
  } else {
    sum = sum + Number(line);
  }
}
if (sum > max) {
  max = sum;
}

console.log(max);
